src/Microsoft.AspNet.Server.Kestrel/Infrastructure/MemoryPoolIterator2.cs

```diff
@@ -439,7 +441,35 @@ public int Seek(Vector<byte> byte0Vector, Vector<byte> byte1Vector, Vector<byte>
             }
         }
 
-        private static int FindFirstEqualByte(Vector<byte> byteEquals)
+        /// <summary>
+        /// Find 
+        /// </summary>
+        /// <param name="byteEquals"></param>
+        /// <returns>The first index of the result vector if found otherwise return int.MaxValue</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
No )-; it would not Inline. So I added the attr. 
A lot of very small functions will inline will inline  
like `public bool IsDefault => _block == null` 

But If  look to the code there are a lot of small functions that can make a big difference
Like peek, peeklong, isend, take. 

They can be split up in a peek and a peeknext block. So the inline code will be small